src/realm/group.cpp

```diff
@@ -300,10 +300,19 @@ int Group::get_committed_file_format_version() const noexcept
     return m_alloc.get_committed_file_format_version();
 }
 
+std::optional<int> fake_target_file_format;
+
+void _impl::GroupFriend::fake_target_file_format(const std::optional<int>& format) noexcept
+{
+    ::fake_target_file_format = format;
+}
 
 int Group::get_target_file_format_version_for_session(int current_file_format_version,
                                                       int requested_history_type) noexcept
 {
+    if (fake_target_file_format) {
```

Comment:
I think we should define that the "target_file_format" is the first element in "accepted_versions". Then Group can get this value from Backup handler. We can avoid this double bookkeeping.